### ISSUE TITLE
[Suggestion] Make ThreadState independent from ChannelState

### ISSUE THREAD

**Issue by workspace** [GetStream/stream-chat-android#3956]
## Intro
until v5.6.1, ThreadState is heavily depends on ChannelState. In my case,
- I use offline plugin with non-persistence. (I don't use ui from stream-chat-android)
- I implemented screen for thread unlike MessageList in stream-chat-android-compose.
- When user clicked notification, I send user to thread screen directly.
- (v4, v5 both) Initialization of [ThreadState](https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/state/StateRegistry.kt#L114-L117) and [ThreadLogic](https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/logic/internal/LogicRegistry.kt#L108-L111) needs message in MessageRepository to get channelType and channelId.
- (v4) I resolved this issue with calling loadMessageById to insert message in db  before getting ThreadController.
- (v5) I tried to resolve with the same way, however, loadMessageById in thread screen make unexpected changes to message list of ChannelState. (maybe due to recently changes with arround id pagination)

In conclusion, I think it is impossible to implement thread screen with v5 sdk.

## Implementation
https://github.com/GetStream/stream-chat-android/compare/develop...workspace:independent-thread-state
- New ThreadState implementation is inspired by ChannelMutableState, ChannelLogic, ChannelStateLogic. 
- ThreadState has message list which is consist of parentMessage and replies. 
- In EventHandler, active threads receive events to handle then update state.

Now, ThreadState needs only parentId to initialize!

## Optimization
- New ThreadState implementation do not use db intentionally because I don't know exactly how to use it safely.  Can I get some help or advice to use it properly?
- Add interface to clear spectific thread in StateRegistry and LogicRegistry via ChatClient.
